app/controllers/verify/finance_controller.rb

```diff
@@ -17,20 +19,39 @@ def create
 
     private
 
+    def step
+      @_step ||= Idv::FinancialsStep.new(
+        idv_finance_form: idv_finance_form,
+        idv_session: idv_session
+      )
+    end
+
+    def step_params
+      params.require(:idv_finance_form).permit(:finance_type, *Idv::FinanceForm::FINANCE_TYPES)
+    end
+
+    def confirm_step_needed
+      redirect_to verify_phone_path if idv_session.financials_confirmation.try(:success?)
+    end
+
     def render_form
-      if finance_params[:finance_type] == 'ccn'
+      if step_params[:finance_type] == 'ccn'
         render :new
       else
         render 'verify/finance_other/new'
       end
     end
 
-    def idv_finance_form
-      @_idv_finance_form ||= Idv::FinanceForm.new(idv_session.params)
+    def track_idv_event
+      result = {
+        success: idv_session.financials_confirmation.success?
+      }
+
+      analytics.track_event(Analytics::IDV_FINANCE_CONFIRMATION, result)
```

Comment:
> If we later decide that we don't need the vendor validation, or need to temporarily disable it, which version of the code would make it easier to do that? Or what if we need to add a third type of validation?

Those are good questions, and the very ones I have been asking myself.

I agree with this statement:

> I think it's cleaner and more readable to have each type of validation as a separate class

which is why I am proposing this new `Step` class as the top-level model. We have `StepController` which now operates on a `Step` object.

You wrote:

> any validations that need to be performed while submitting the form can be called by the Form Object since that is the point of entry. This makes the code more modular and easy to change.

I am suggesting we switch the `Step` to be the point of entry. A user completes a Step, which includes the Form. But the Form validation is only one part of the Step; another part is the vendor confirmation.

I think I've hit your concerns:

* multiple validations are handled by separate classes
* validations are grouped together under a single parent `Step` class
* analytics can reflect both form validations and vendor validations

Are we arguing about the name `Form` vs `Step`?